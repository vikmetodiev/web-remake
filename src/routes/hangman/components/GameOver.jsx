import React from 'react';
import winImage from '../img/win.png';
import loseImage from '../img/7.png';
import styled from 'styled-components';

const Text = styled.p`
  display: ${props => (props.newGame ? 'none' : 'block')};
  font-size: 26px;
  max-width: 200px;
  margin: 0;
  padding-right: 40px;
  &:after {
    content: '';
    width: ${props => (props.isLose ? '180px' : '100px')};
    height: ${props => (props.isLose ? '180px' : '158px')};
    background-repeat: no-repeat;
    display: inline-block;
    background-size: 100%;
    margin: 10px;
    position: relative;
    left: 100px;
    background-image: ${props =>
      props.isLose ? `url(${loseImage})` : `url(${winImage})`};
  }
`;

const GameOver = props => <Text {...props}>{props.text}</Text>;

export default GameOver;
