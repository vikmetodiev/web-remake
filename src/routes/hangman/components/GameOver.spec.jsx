import React from 'react';

import toJson from 'enzyme-to-json';
import GameOver from './GameOver';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('Game Over', () => {
  const text = 'Snap!';

  it('should render the component', () => {
    const wrapper = shallow(<GameOver text={text} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
