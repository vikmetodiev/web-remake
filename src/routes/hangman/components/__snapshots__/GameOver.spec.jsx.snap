// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Over should render the component 1`] = `
<styled.p
  text="Snap!"
>
  Snap!
</styled.p>
`;
